hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetContentsHandler.java

```diff
@@ -77,13 +78,20 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
         final var fileStore = context.createStore(ReadableFileStore.class);
         final var op = query.fileGetContentsOrThrow();
         final var responseBuilder = FileGetContentsResponse.newBuilder();
-        final var file = op.fileIDOrElse(FileID.DEFAULT);
+        final var file = op.fileIDOrThrow();
 
         final var responseType = op.headerOrElse(QueryHeader.DEFAULT).responseType();
         responseBuilder.header(header);
         if (header.nodeTransactionPrecheckCode() == OK && responseType != COST_ANSWER) {
             final var optionalInfo = contentFile(file, fileStore);
-            optionalInfo.ifPresent(responseBuilder::fileContents);
+
+            if (optionalInfo.isEmpty()) {
+                responseBuilder.header(header.copyBuilder()
+                        .nodeTransactionPrecheckCode(INVALID_FILE_ID)
+                        .build());
+            } else {
+                optionalInfo.ifPresent(responseBuilder::fileContents);
```

Comment:
Instead of lambda syntax (which is costly), we've already verified that the optionalInfo is present and not empty, we can just call `responseBuilder.fileContents(optionalInfo.get());` directly.

Reply:
+1

Reply:
done thanks